manifests/jail.pp

```diff
@@ -34,80 +34,69 @@
 # @param jail_name
 #   The jails name.
 #
+# @param jail_base
+#   The jail's full path.
+#
 # @param user
-#   The user that will own the corresponding home directory in the jail, giving
-#   the user a place to land. Also sets user ownership for `/incoming`.
+#   The main user that will own the corresponding home directory in the jail,
+#   giving the user a place to land. Also sets user ownership for `/incoming`.
 #
 # @param group
 #   The group that will own the corresponding home directory in the jail,
 #   giving the user a place to land. Also sets group ownership for `/incoming`.
 #
+# @param sub_dirs
+#   This directory structure is enforced in the users Home.
+#
+# @param merge_subdirs
+#   Merge sub_dirs with the default sub_dirs?
+#
 # @param match_group
 #   Set the group that SSHd will look for when redirecting users to the jail.
-#   Useful for shared jails. Defaults to the value of `group`.
+#   Useful for shared jails.
 #
 # @param password_authentication
-#   OpenSSH Password Authentication
+#   Can the user login with a password? Public key authentication is generally
+#   recommended and has to be configured outside of the scope of this module.
 #
 define sftp_jail::jail (
-  $jail_name               = $name,
-  $user                    = undef,
-  $group                   = undef,
-  $match_group             = undef,
-  $password_authentication = 'no',
+  Sftp_jail::File_name      $jail_name               = $name,
+  Stdlib::Absolutepath      $jail_base               = "${sftp_jail::chroot_base}/${jail_name}",
+  Sftp_jail::User_name      $user                    = $name,
+  Sftp_jail::User_name      $group                   = $user,
```

Comment:
where is `$group` coming from?

Reply:
The group had `$user` ad default value since before my changes. I just simplified it this commit: https://github.com/voxpupuli/puppet-sftp_jail/pull/71/commits/0dbefc852efefdd0669775cc7e21df5496a62895. The PR for it: https://github.com/voxpupuli/puppet-sftp_jail/pull/75

The users default value is (now) explained in this commit: https://github.com/voxpupuli/puppet-sftp_jail/pull/71/commits/223758944d9838a24d524cb46c6de64081a29c00. The PR for it: https://github.com/voxpupuli/puppet-sftp_jail/pull/80